packages/inter-protocol/src/priceAggregatorChainlink.js

```diff
@@ -729,66 +708,64 @@ const start = async (zcf, privateArgs) => {
 
       return zcf.makeInvitation(offerHandler, INVITATION_MAKERS_DESC);
     },
-    /** @param {string} oracleAddr */
-    deleteOracle: async oracleAddr => {
-      const records = keyToRecords.get(oracleAddr);
-      for (const record of records) {
-        records.delete(record);
-      }
+    /** @param {string} oracleId */
+    deleteOracle: async oracleId => {
+      keyToRecords.delete(oracleId);
 
-      oracleStatuses.delete(oracleAddr);
+      oracleStatuses.delete(oracleId);
 
       // We should remove the entry entirely, as it is empty.
-      keyToRecords.delete(oracleAddr);
+      keyToRecords.delete(oracleId);
     },
 
     /**
-     * @param {string} oracleAddr Bech32 of oracle operator smart wallet
+     * @param {string} oracleId unique per contract instance
      */
-    async initOracle(oracleAddr) {
-      assert.typeof(oracleAddr, 'string');
+    async initOracle(oracleId) {
+      assert.typeof(oracleId, 'string');
       /** @type {OracleRecord} */
       const record = { querier: undefined, lastSample: 0 };
 
-      /** @type {Set<OracleRecord>} */
+      /** @type {SetStore<OracleRecord>} */
       let records;
-      if (keyToRecords.has(oracleAddr)) {
-        records = keyToRecords.get(oracleAddr);
+      if (keyToRecords.has(oracleId)) {
+        records = keyToRecords.get(oracleId);
+        records.add(record);
       } else {
-        records = new Set();
-        keyToRecords.init(oracleAddr, records);
-
-        const oracleStatus = makeOracleStatus(
-          /* startingRound = */ getStartingRound(oracleAddr),
-          /* endingRound = */ ROUND_MAX,
-          /* lastReportedRound = */ 0n,
-          /* lastStartedRound = */ 0n,
-          /* latestSubmission = */ 0n,
-          /* index = */ oracleStatuses.getSize(),
+        records = makeScalarBigSetStore(`${oracleId} records`, {
+          durable: true,
```

Comment:
`records` need not be mutable, right? It need not retain identity when the relevant contents change? So I suggest a CopySet instead.